projects/RabbitMQ.Client/client/impl/ChannelBase.cs

```diff
@@ -1785,6 +1825,63 @@ await tokenRegistration.DisposeAsync()
             }
         }
 
+        // NOTE: this method is internal for its use in this test:
+        // TestWaitForConfirmsWithTimeoutAsync_MessageNacked_WaitingHasTimedout_ReturnFalse
+        internal void HandleAckNack(ulong deliveryTag, bool multiple, bool isNack)
```

Comment:
Any reason this method doesn't use async and a cancellation token? Then the semaphore could be awaited and cancelled. It seems possible to flow through the task since the method is used in `DispatchCommandAsync` which already follows a similar pattern

Reply:
Yes, I was wondering about this while I was working on the publisher confirmation test code. I'll take a look today.

Reply:
See https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1675